state-sync/state-sync-v2/state-sync-multiplexer/src/lib.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+#![allow(dead_code)]
```

Comment:
I don't see any dead code in here, probably can remove that?

Also, do we really need another crate for this multiplexer?  How many crates do we have now?

Reply:
Regarding the `dead_code` this is a result of what's still to come (e.g., having the node config support this value) so I'll just leave it in anticipation 😄 

> I just want us to think about only creating new crates for components that specifically are used in different combinations of components

Generally, I'd agree with you 😄 The only reason I made this its own crate was so that once we remove state sync v1, I'll be able to delete this crate entirely (which seems a little cleaner than cutting out files manually). For the moment, it also simplifies interfaces (e.g., diem node only depends about the multiplexer crate, as opposed to both version of state sync). This will all hopefully be removed soon though!